### PR TITLE
Enhancement/creating condition table dynamically

### PR DESCRIPTION
**Pull request by Yagnik56**
This PR is created to make condition table dynamically with no limitation over number of factors in factor table.

### CODE REVIEW

File: frontend/projects/upgrade/src/app/core/experiment-design-stepper/experiment-design-stepper.service.ts [hunk 1]
```diff
@@ -294,45 +302,76 @@ export class ExperimentDesignStepperService {
...
+    conditionData.forEach((level) => {
+      levels.push({ id: level.id, name: level.name });
+    });
+    return levels;
+  }
+
+  createConditionString(conditionData: FactorLevelData[]) {
+    let alias: string;
+    let count = 0;
+    conditionData.forEach((level) => {
```
Comment (danoswaltCL) [inline frontend/projects/upgrade/src/app/core/experiment-design-stepper/experiment-design-stepper.service.ts:364]: Couple nitpicky things, ignore me if you like. fyi `Array.forEach` comes with an index argument: `conditionData.forEach((level, index)`, you don't have to track it manually with `count`.

Also I don't know if there's a possibility if this, but I notice that if `conditionData` is empty somehow, this function will return `undefined` for alias, which I doubt is a happy result. Maybe just initialize as empty string?

File: frontend/projects/upgrade/src/app/core/experiment-design-stepper/experiment-design-stepper.service.ts [hunk 2]
```diff
@@ -294,45 +302,76 @@ export class ExperimentDesignStepperService {
...
-      });
+      const tableRow: FactorialConditionTableRowData = {
+        id: uuidv4(), // TODO: maybe not the right place?
+        levels: conditionLevelsData,
+        condition: conditions,
+        alias: '',
+        weight: '0.0',
+        include: true,
+      };
+      console.log("tableRow.condition.split(';').length");
```
Comment (danoswaltCL) [inline frontend/projects/upgrade/src/app/core/experiment-design-stepper/experiment-design-stepper.service.ts:319]: oops console logs...

Comment (Yagnik56) [reply frontend/projects/upgrade/src/app/core/experiment-design-stepper/experiment-design-stepper.service.ts]: Oops sorry missed that, it's removed now

File: frontend/projects/upgrade/src/app/core/experiment-design-stepper/experiment-design-stepper.service.ts (hunk 1 above)
Comment (Yagnik56) [reply frontend/projects/upgrade/src/app/core/experiment-design-stepper/experiment-design-stepper.service.ts]: resolved both

**Review by jreddig — APPROVED** [review]

**Pull request merged**